### PR TITLE
update DQN

### PR DESCRIPTION
**Pull request by MorganeLu**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/deep-learning/DQN.md [hunk 1]
```diff
@@ -13,10 +13,367 @@ kernelspec:
...
+- Often it is considered a real-number $r_k \in \mathbb{R}$ or an integer $r_k \in \mathbb{Z}$.
+- The reward function (interpreter) may be naturally given or is a design degree of freedom (i.e., can be manipulated).
+- It fully indicates how well an RL agent is doing at step $k$.
+- The agent’s task is to maximize its reward over time.
+
+```{example}
+If we want the machine to flip a pancake:
+- Pos. reward: catching the 180◦ rotated pancake
+- Neg. reward: droping the pancake on the floor
+```
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/deep-learning/DQN.md:56]: maybe you mean {note}?

Comment (reveurmichael) [inline open-machine-learning-jupyter-book/deep-learning/DQN.md:56]: maybe you mean {note}?

File: open-machine-learning-jupyter-book/deep-learning/DQN.md [hunk 2]
```diff
@@ -13,10 +13,367 @@ kernelspec:
...
+            wandb.log({'Reward': total_reward})
+
+
+def main():
+    env = gym.make('CartPole-v1')
+    agent = Agent(env)
+    agent.train(max_episodes=1000)
+
+if __name__ == "__main__":
+    main()
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/deep-learning/DQN.md:358]: The code here and the text example is not the same game?

File: open-machine-learning-jupyter-book/deep-learning/DQN.md [hunk 3]
```diff
@@ -13,10 +13,367 @@ kernelspec:
...
+- It fully indicates how well an RL agent is doing at step $k$.
+- The agent’s task is to maximize its reward over time.
+
+```{example}
+If we want the machine to flip a pancake:
+- Pos. reward: catching the 180◦ rotated pancake
+- Neg. reward: droping the pancake on the floor
+```
+
+Rewards can have many different flavors and are highly depending on the given problem:
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/deep-learning/DQN.md:58]: depending -> dependent

File: open-machine-learning-jupyter-book/deep-learning/DQN.md (hunk 1 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/DQN.md]: Yes, done.

File: open-machine-learning-jupyter-book/deep-learning/DQN.md (hunk 3 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/DQN.md]: Done.

File: open-machine-learning-jupyter-book/deep-learning/DQN.md (hunk 2 above)
Comment (MorganeLu) [reply open-machine-learning-jupyter-book/deep-learning/DQN.md]: Yes, I changed the code part in the maze game.

**Review by reveurmichael — APPROVED** [review]

**Pull request merged**